augraphy/augmentations/delaunay.py

```diff
@@ -0,0 +1,230 @@
+
+- Delaunay Tessellation: https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.Delaunay.html
+
+- Perlin Noise: https://iq.opengenus.org/perlin-noise/
+
+*********************************
+
+"""
+import random
+import warnings
+
+import cv2
+import numpy as np
+from scipy import ndimage
+
+from augraphy.base.augmentation import Augmentation
+from augraphy.utilities.meshgenerator import Noise
+from augraphy.utilities.slidingwindow import PatternMaker
+
+# from scipy.spatial import Delaunay
+
+warnings.filterwarnings("ignore")
+
+
+class DelaunayTessellation(Augmentation):
+    """
+    The Delaunay Tessellation is a method of dividing a geometric space
+    into a set of triangles. This implementation generate a Delaunay Tessellation of an image with Perlin Noise by default to create smoother,
+    more organic looking tessellations.
+    The Delaunay Tessellation algorithm is a method of traingulating a set of points in a planar space such that the minimum angle of
+    each triangle is maximized. This ensures that the triangles are as close to equilateral as possible.
+    The Delaunay Tessellation is defined as the triangulation of a set of points such that no point is inside the circumcircle of any triangle.
+    The algorithm works by iteratively adding points to the triangulation and re-triangulating the set of points after each point is added.
+    The algorithm ensures that the triangulation remains a Delaunay tessellation by checking for the Delaunay condition after each point is added.
+    The Delaunay Condition states that the circumcircle of each triangle in the triangulation must contain no other points in its interior.
+    The class inherits methods and properties from the Augmentation base class.
+
+    :param n_points: Range for the number of triangulating points
+    :type n_points: tuple (int)
+    :param n_horizontal_points: Range for the number of points in the horizontal edge, default = 200
+    :type n_horizontal_points: tuple (int)
+    :param n_vertical_points: Range for the number of points in the vertical edge, default = 200
+    :type n_vertical_points: tuple (int)
+    :param perlin: if True, Perlin Noise is added to the distance between each point and its closest random point to
+    create a smoother, more organic looking tessellation. Default is True.
+    :type perlin: bool
+    :param ws: patch height and width, default value is 200
+    :type ws: int
+    :param p: The probability of applying the augmentation to an input image. Default value is 1.0
+    :type p: float
+
+    """
+
+    def __init__(
+        self,
+        width=500,
+        height=500,
+        n_points=(400, 800),
+        n_horizontal_points=(1, 50),
+        n_vertical_points=(1, 50),
+        perlin=True,
+        ws=200,
+        color_background=(200, 255),
+        p=1,
+    ):
+        super().__init__(p=p)
+        self.width = width
+        self.height = height
+        self.n_points = random.randint(
```

Comment:
So we need a different approach here. The actual randomized parameter should be initialized during the `__call__` so that the parameter value will be different from each call.